java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -22,18 +20,20 @@ import kotlinx.coroutines.sync.withLock
  */
 abstract class AbstractArcHost : ArcHost {
     private val hostMutex = Mutex()
-    private var particles: MutableList<KClass<out Particle>> by
+    private var particles: MutableList<ParticleIdentifier> by
         guardWith(hostMutex, mutableListOf())
 
-    override suspend fun registerParticle(particle: KClass<out Particle>) {
+    override fun hostId() = this::class.java.canonicalName!!
```

Comment:
is `canonicalName` really nullable?

Reply:
IntelliJ says it has an @Nullable annotation, yeah, it surprised me too.